### PR TITLE
CDAP-2811: Improve Hive version detection

### PR DESCRIPTION
**Pull request by jwang47**
https://issues.cask.co/browse/CDAP-2811
http://builds.cask.co/browse/CDAP-DUT2070

* Works in standalone and distributed

### CODE REVIEW

File: cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java [hunk 1]
```diff
@@ -191,26 +179,16 @@ public static HiveSupport checkHiveSupport(ClassLoader hiveClassLoader) {
...
+      } else if (hiveVersion.startsWith("0.13.")) {
+        return HiveSupport.HIVE_13;
       }
+
       throw new RuntimeException("Hive distribution not supported. Set the configuration '" +
-                                 Constants.Explore.EXPLORE_ENABLED +
-                                 "' to false to start up without Explore.");
+                                   Constants.Explore.EXPLORE_ENABLED +
+                                   "' to false to start up without Explore.");
     } catch (RuntimeException e) {
```
Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java]: does this still applies?

Comment (jwang47) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java]: I think so, since we don't support every version of Hive.

Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java]: but catching a `RuntimeException` and throwing it is meaningless. Why catch and rethrow?

Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java]: In fact, if the `HiveVersionInfo.getVersion()` doesn't throw checked exception, there is no need for a try block here.

Comment (jwang47) [inline cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java]: True, I misread and thought your comment was on the throw line. Fixed.

**Comment by jwang47** [comment]
Reviewed in https://github.com/caskdata/cdap/pull/3004

**Pull request merged**